genrl/deep/agents/a2c/a2c.py

```diff
@@ -98,19 +103,29 @@ def create_model(self) -> None:
         """
         Creates actor critic model and initialises optimizers
         """
-        input_dim, action_dim, discrete, action_lim = get_env_properties(
-            self.env, self.network_type
-        )
-
+        if isinstance(self.network, str):
+            input_dim, action_dim, discrete, action_lim = get_env_properties(
+                self.env, self.network
+            )
+            self.ac = get_model("ac", self.network)(
+                input_dim, action_dim, self.layers, "V", discrete, action_lim=action_lim
+            ).to(self.device)
+
+        else:
+            self.ac = self.network
+            action_dim = self.network.action_dim
+
+        assert "actor" and "critic" in dir(
+            self.ac
+        ), "network must contain actor and critic attributes"
+        assert "get_value" and "get_action" in dir(
+            self.ac
+        ), "network must contain get_value and get_action methods"
```

Comment:
Can be moved into the else statement. May we could do without asserting for `actor` and `critic`. It's best to have as few assert statements as possible.